### DISCUSSION TITLE
Conversion to TypeScript

### DISCUSSION THREAD

**Discussion by kevinhillinger** [wdfox/DevOpsWithAPIM#14] [General]
## local setup

```
npm install -g typescript
```

## project setup

We're going to need the following (minimum) things to convert to TypeScript:

1. initialize a ts project ```npx tsc --init```
2. add a .gitignore file to the repo for typescript
3. in the tsconfig, set output directory to /dist
4. for each .js file, convert to .ts extension
5. update code in each file

**Comment by kevinmgates** [comment]
Output directory setting in the tsconfig file is line #50

https://www.typescriptlang.org/tsconfig#outDir

....and it's enabled. All. The. Red. Lol.
![image](https://user-images.githubusercontent.com/21062553/151234376-bb5729b6-bfe5-4d2f-b56b-5efe9524ef64.png)

**Reply by kevinmgates** [reply]
Helpful video on TypeScript in 100 seconds: https://www.youtube.com/watch?v=zQnBQ4tB3ZA

A longer, intro to TS is here....also really good: https://www.youtube.com/watch?v=ahCwqrYpIuM

**Comment by kevinmgates** [comment]
No more red! I converted the file to typescript!
![image](https://user-images.githubusercontent.com/21062553/151240932-8aa15a3b-29a5-4bf0-9c58-d30853ffc0c7.png)

**Reply by kevinhillinger** [reply]
Excellent! We'll go through it and you can submit the PR (if you haven't already).
